src/include/distributed/backend_data.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * backend_data.h
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef BACKEND_DATA_H_
+#define BACKEND_DATA_H_
+
+
+#include "datatype/timestamp.h"
+#include "nodes/pg_list.h"
+#include "storage/s_lock.h"
+
+
+/*
+ * Citus identifies a distributed transaction with a triplet consisting of
+ *
+ *  -  initiatorNodeIdentifier: A unique identifier of the node that initiated
+ *     the distributed transaction
+ *  -  transactionNumber: A locally unique identifier assigned for the distributed
+ *     transaction on the node that initiated the distributed transaction
+ *  -  timestamp: The current timestamp of distributed transaction initiation
+ *
+ */
+typedef struct DistributedTransactionId
+{
+	uint64 initiatorNodeIdentifier;
+	uint64 transactionNumber;
+	TimestampTz timestamp;
+} DistributedTransactionId;
+
+
+/*
+ * Each backend's active distributed transaction information is tracked via
+ * BackendData on the shared memory.
+ */
+typedef struct DistributedTransactionBackendData
+{
+	Oid databaseId;
+	int processId;
+	slock_t mutex;
+	DistributedTransactionId transactionId;
+} BackendData;
+
+
+extern void InitializeBackendManagement(void);
+extern void InitializeBackendData(void);
+extern void UnSetBackendData(void);
+extern DistributedTransactionId * GetCurrentDistributedTransctionId(void);
+extern void AssignBackendData(void);
```

Comment:
I think `UnSetBackendData`, and  `AssignBackendData` should still refer to transaction identifiers.